### PR TITLE
Use service case type id in the startEvent call for the attachScannedDocsWithOcr event

### PR DESCRIPTION
**Pull request by aliveni**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/BPS-1180

### Change description ###
Retrieve the service case before starting the "attachScannedDocsWithOcr" event so that the service case type id can be fetched and used in the startEvent call.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/AttachCaseCallbackService.java [hunk 1]
```diff
@@ -447,14 +447,17 @@ private void updateSupplementaryEvidence(
             callBackEvent.exceptionRecordId
         );
 
+        CaseDetails targetCase = ccdApi.getCase(targetCaseCcdRef, callBackEvent.exceptionRecordJurisdiction);
```
Comment (nitinprabhu) [inline src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/AttachCaseCallbackService.java:450]: if this call fails will the caseworker shown an error screen. If this call fails they should be shown with a message to retry.

Comment (nitinprabhu) [reply src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/AttachCaseCallbackService.java:450]: Also test case is missing when this call throws an exception

File: src/test/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/CcdCaseUpdaterTest.java [hunk 2]
```diff
@@ -466,7 +478,7 @@ void updateCase_should_handle_feign_exception_from_start_event() {
             anyString(),
             anyString()
         ))
-            .willThrow(new FeignException.MethodNotAllowed("Msg", mock(Request.class),  "Body".getBytes()));
+            .willThrow(new FeignException.MethodNotAllowed("Msg", mock(Request.class), "Body".getBytes()));
```
Comment (nitinprabhu) [inline src/test/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/CcdCaseUpdaterTest.java:481]: not needed change ?

File: src/test/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/CcdCaseUpdaterTest.java [hunk 3]
```diff
@@ -534,7 +548,7 @@ void updateCase_should_handle_exception_when_start_event_returns_not_found_respo
             anyString()
         ))
             .willThrow(
-                    new FeignException.NotFound("case not found",  mock(Request.class), "Body".getBytes())
+                new FeignException.NotFound("case not found", mock(Request.class), "Body".getBytes())
```
Comment (nitinprabhu) [inline src/test/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/CcdCaseUpdaterTest.java:551]: same here formatting change not needed ?

File: src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/AttachCaseCallbackService.java [hunk 4]
```diff
@@ -447,14 +447,17 @@ private void updateSupplementaryEvidence(
...
         ServiceConfigItem serviceConfigItem = getServiceConfig(exceptionRecordDetails);
         ProcessResult processResult = ccdCaseUpdater.updateCase(
             callBackEvent.exceptionRecord,
             serviceConfigItem,
             ignoreWarnings,
             callBackEvent.idamToken,
             callBackEvent.userId,
-            targetCaseCcdRef
+            targetCaseCcdRef,
+            targetCase.getCaseTypeId()
```
Comment (doncem) [inline src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/AttachCaseCallbackService.java:460]: isn't this available from `callBackEvent` (same as `targetCaseCcdRef`)?

Comment (aliveni) [reply src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/AttachCaseCallbackService.java:460]: `targetCaseCcdRef` is for the "search case reference type"  which can be either CCD case reference or external case reference

File: src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/AttachCaseCallbackService.java (hunk 1 above)
Comment (aliveni) [reply src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/AttachCaseCallbackService.java:450]: I'll update the pr

Comment (aliveni) [reply src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/AttachCaseCallbackService.java:450]: When the ccd call fails, the error will be displayed to the caseworker. see https://github.com/hmcts/bulk-scan-orchestrator/blob/a0b8a51ac54e9a57863b7ca7c5c3483596a90bee/src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/AttachCaseCallbackService.java#L235-L248

I've updated the integration tests for invalid case reference and case not found scenarios.

**Pull request merged**